src/Voron/Platform/PalHelper.cs

```diff
@@ -37,5 +35,17 @@ public static unsafe void ThrowLastError(int lastError, string msg)
 
             throw new InvalidOperationException(txt);
         }
+
+        public static unsafe string GetNativeErrorString(int lastError, string msg, out ErrnoSpecialCodes errnoSpecialCodes)
+        {
+            const int maxNativeErrorStr = 256;
+            var buf = stackalloc byte[maxNativeErrorStr];
+
+            var size = Pal.rvn_get_error_string(lastError, buf, maxNativeErrorStr, out var specialErrnoCodes);
```

Comment:
In order to produce good error, we need both the special error code _and_ the error code